frontends/web/src/components/dropdown/mobile-fullscreen-selector.tsx

```diff
@@ -0,0 +1,171 @@
+  renderOptions: (option: TOption<T>) => ReactNode;
+  value: IsMulti extends true ? TOption<T>[] : TOption<T>;
+  onChange: (newValue: IsMulti extends true ? TOption<T>[] : TOption<T>, actionMeta: ActionMeta<TOption<T>>) => void;
+  isMulti?: boolean;
+  isOpen?: boolean;
+  onOpenChange?: (isOpen: boolean) => void;
+}
+
+export const MobileFullscreenSelector = <T, IsMulti extends boolean = false>({
+  title = 'Select',
+  options,
+  renderOptions,
+  value,
+  onChange,
+  isMulti,
+  isOpen: controlledIsOpen,
+  onOpenChange,
+}: Props<T, IsMulti>) => {
+  const [internalIsOpen, setInternalIsOpen] = useState(false);
+  const [searchText, setSearchText] = useState('');
+  const { t } = useTranslation();
+
+  const isControlled = controlledIsOpen !== undefined && onOpenChange !== undefined;
+  const isOpen = isControlled ? controlledIsOpen : internalIsOpen;
+
+  useEffect(() => {
+    if (!isOpen) {
+      setSearchText('');
+    }
+  }, [isOpen]);
+
+  const handleOpen = () => {
+    if (isControlled && onOpenChange) {
+      onOpenChange(true);
+    } else {
+      setInternalIsOpen(true);
+    }
+  };
+
+  const handleClose = () => {
+    if (isControlled && onOpenChange) {
+      onOpenChange(false);
+    } else {
+      setInternalIsOpen(false);
+    }
+  };
+
+  const handleBackButtonClick = (e: React.MouseEvent) => {
+    e.stopPropagation();
+    handleClose();
+  };
+
+  const isSelected = (option: TOption<T>) => {
+    if (isMulti) {
+      return (value as TOption<T>[]).some((v) => v.value === option.value);
+    }
+    return (value as TOption<T>).value === option.value;
+  };
+
+  const handleSelect = (option: TOption<T>, e: React.MouseEvent) => {
+    e.stopPropagation();
+    if (isMulti) {
+      const currentValues = value as TOption<T>[];
+      const isCurrentlySelected = currentValues.some((v) => v.value === option.value);
+
+      let newValue: TOption<T>[];
+      if (isCurrentlySelected) {
+        newValue = currentValues.filter((v) => v.value !== option.value);
+      } else {
+        newValue = [...currentValues, option];
+      }
+
+      onChange(newValue as any, { action: isCurrentlySelected ? 'deselect-option' : 'select-option', option } as any);
+    } else {
+      onChange(option as any, { action: 'select-option', option } as any);
+      handleClose();
+    }
+  };
+
+  const displayValue = isMulti
+    ? (value as TOption<T>[]).map(v => v.label).join(', ') || 'Select...'
+    : (value as TOption<T>).label || 'Select...';
+
+  const filteredOptions = options.filter(option =>
+    option.label.toLowerCase().includes(searchText.toLowerCase())
+  );
+
+  return (
+    <div className={styles.dropdownContainer}>
+      <button
+        className={styles.mobileDropdownTrigger}
+        onClick={handleOpen}
+        type="button"
+      >
+        <span className={styles.mobileDropdownValue}>{displayValue}</span>
+        <div className={styles.dropdown} />
+      </button>
+
+      {isOpen && (
```

Comment:
Currently if you use native back on Android it goes to mobilesettings, but it would be nice if it goes to general settings.

Could you intercept and prevent the native back behavior and close the fullscreen selector instead?

untested but something like this I believe:

```jsx
<UseBackButton handler={() => {
  handleBackButtonClick();
  return false;
}} />
```